R/render.R

```diff
@@ -333,6 +337,76 @@ render <- function(input,
     }, add = TRUE)
     env$metadata <- yaml_front_matter
 
+    # hook up output_source to evaluate if specified
```

Comment:
Again, I was thinking that the html_notebook format would have an output_source and the concept would remain scoped there (i.e. render wouldn't be aware of it). It very well could be that this isn't possible, but I think it's cleaner if we keep this concept associated with html_notebook. If you are uncertain we can just discuss realtime in Austin on Monday.

It should be the case that anything render can do in terms of hooks a format can do. If this isn't the case perhaps we could make whatever changes to `render` required to make this so.